src/components/message/message.pw.tsx

```diff
@@ -0,0 +1,333 @@
+    test(`should apply expected styling when transparent is ${boolVal}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<MessageComponent transparent={boolVal} />);
+
+      await expect(messagePreview(page)).toHaveCSS(
+        "background-color",
+        backgroundColor
+      );
+    });
+  });
+
+  ([
+    [true, 50],
+    [false, 20],
+  ] as [boolean, number][]).forEach(([boolVal, paddingVal]) => {
+    test(`should check showCloseIcon when it's ${boolVal} for Message component`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<MessageComponent showCloseIcon={boolVal} />);
+
+      if (boolVal === true) {
+        await expect(messageDismissIcon(page)).toBeVisible();
+        await expect(messageContent(page)).toHaveCSS(
+          "padding",
+          `15px ${paddingVal}px 15px 20px`
+        );
+      } else {
+        await expect(messageDismissIcon(page).getAttribute("disabled"))
+          .toBeTruthy;
+        await expect(messageContent(page)).toHaveCSS(
+          "padding",
+          `15px ${paddingVal}px`
+        );
+      }
+    });
+  });
+
+  testData.forEach((ariaLabel) => {
+    test(`should check closeButtonAriaLabel as ${ariaLabel} for Message component`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<MessageComponent closeButtonAriaLabel={ariaLabel} />);
+
+      await expect(messageDismissIconButton(page)).toHaveAttribute(
+        "aria-label",
+        ariaLabel
+      );
+    });
+  });
+
+  test("should call onDismiss callback when a click event is triggered", async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <MessageComponent
+        onDismiss={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+    await messageDismissIcon(page).click();
+    expect(callbackCount).toBe(1);
+  });
+
+  test("should render message component with expected border radius styling", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<MessageComponent />);
+
+    await expect(messagePreview(page)).toHaveCSS("border-radius", "8px");
+  });
+
+  // Unable to run this test due to the Message component having a tabIndex of -1.
+  // cypress-plugin-tab does not allow tabbing on elements with an tabIndex of -1,
+  // however this test has been kept incase the bug is ever addressed. https://github.com/kuceb/cypress-plugin-tab/issues/18
+
+  test.skip("should focus icon button when open is true for Message component and tab key is pressed", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<MessageComponentWithRef />);
+    await buttonPreview(page).click();
+    await expect(messagePreview(page)).toBeFocused();
+    await page.keyboard.press("Tab");
+    await expect(messageDismissIconButton(page)).toBeFocused();
+  });
+});
+
+test.describe("Accessibility tests for Message component", () => {
+  (["info", "error", "success", "warning"] as any[]).forEach((variant) => {
```

Comment:
type "any" can be replaced by the proper type for ["info", "error", "success", "warning"], which is MessageProps["variant"]